packages/opds-web-client/src/components/__tests__/Root-test.tsx

```diff
@@ -38,17 +63,38 @@ const setCollectionAndBookPromise = new Promise((resolve, reject) => {
   });
 });
 const mockSetCollectionAndBook = stub().returns(setCollectionAndBookPromise);
+const fulfillBook = async (url: string): Promise<Blob> => new Blob();
+const updateBook = async (url: string): Promise<BookData> => book;
+const indirectFulfillBook = async (
+  url: string,
+  type: string
+): Promise<string> => "test value";
+const setPreference = () => null;
 
 describe("Root", () => {
   it("shows skip navigation link", () => {
-    let wrapper = shallow(<Root />);
+    let wrapper = shallow(
+      <Root
+        fulfillBook={fulfillBook}
+        indirectFulfillBook={indirectFulfillBook}
+        updateBook={updateBook}
+        setPreference={setPreference}
+      />
+    );
```

Comment:
I think I missed this but why are these props not all mandatory?

Reply:
I assume you mean "now all mandatory"? I made them required because `Root` was implicitly expecting them to be there and this seemed more logical than handling cases where they weren't defined. If you think they should be optional, I can make them that way but then we'll have to figure out how `Root` should work if they aren't defined, which I'm not really sure what the expected behavior would be.